File: frontend/src/types/trends.ts
export interface TrendItem {
  month: number;
  total: number;
}

export interface FinancialProfile {
  name: string;
  description: string;
}

export interface TrendResponse {
  top_expenses: TrendItem[];
  top_savings: TrendItem[];
  financial_profile: FinancialProfile;
}

File: backend/schemas/trends.py
from pydantic import BaseModel
from typing import List

class TrendData(BaseModel):
    month: int
    total: float

class FinancialProfile(BaseModel):
    name: str
    description: str
    spotify_link: str

class TrendResponse(BaseModel):
    top_expenses: List[TrendData]
    top_savings: List[TrendData]
    financial_profile: FinancialProfile
